common/lib/xmodule/xmodule/tests/test_split_test_module.py

```diff
@@ -260,29 +265,27 @@ def test_editable_settings(self):
         Test the setting information passed back from editable_metadata_fields.
         """
         editable_metadata_fields = self.split_test_module.editable_metadata_fields
-        self.assertIn(SplitTestDescriptor.display_name.name, editable_metadata_fields)
-        self.assertNotIn(SplitTestDescriptor.due.name, editable_metadata_fields)
-        self.assertNotIn(SplitTestDescriptor.user_partitions.name, editable_metadata_fields)
+        self.assertIn(SplitTestBlock.display_name.name, editable_metadata_fields)
+        self.assertNotIn(SplitTestBlock.due.name, editable_metadata_fields)
+        self.assertNotIn(SplitTestBlock.user_partitions.name, editable_metadata_fields)
 
         # user_partition_id will always appear in editable_metadata_settings, regardless
         # of the selected value.
-        self.assertIn(SplitTestDescriptor.user_partition_id.name, editable_metadata_fields)
+        self.assertIn(SplitTestBlock.user_partition_id.name, editable_metadata_fields)
 
     def test_non_editable_settings(self):
         """
         Test the settings that are marked as "non-editable".
         """
         non_editable_metadata_fields = self.split_test_module.non_editable_metadata_fields
-        self.assertIn(SplitTestDescriptor.due, non_editable_metadata_fields)
-        self.assertIn(SplitTestDescriptor.user_partitions, non_editable_metadata_fields)
-        self.assertNotIn(SplitTestDescriptor.display_name, non_editable_metadata_fields)
+        self.assertIn(SplitTestBlock.due, non_editable_metadata_fields)
+        self.assertIn(SplitTestBlock.user_partitions, non_editable_metadata_fields)
+        self.assertNotIn(SplitTestBlock.display_name, non_editable_metadata_fields)
 
     def test_available_partitions(self):
         """
         Tests that the available partitions are populated correctly when editable_metadata_fields are called
         """
-        self.assertEqual([], user_partition_values.values)
```

Comment:
This is a thread local object and after the conversion the code init-ing this value (in `editable_metadata_fields()`) is sometimes getting executed before this test runs. Is there a way to keep this?

Reply:
I don't see a quick way out either :(

Reply:
Ah, gross.

Consider doing a `patch.object` against `UserPartitionValues.values`. It's a bit hacky, but IMO it's better than creating a potential test-order dependency.

Reply:
Okay, I have added a patch to reset the value of `user_partition_values.values`.